### PR TITLE
[TextureCache][JobMananger][KeyboardLayoutManager][ApplicationMessenger] Kill some more static global singletons.

### PR DESCRIPTION
**Pull request by ksooo**
Kill more of those global singletons, access via Service Broker instead, construct and destruct in a controlled way.

* CJobManager::GetInstance
* CKeyboardLayoutManager::GetInstance
* CApplicationMessenger::GetInstance
* CTextureCache::GetInstance

Runtime-tested on Android and macOS, latest Kodi master

### CODE REVIEW

File: xbmc/ServiceBroker.cpp [hunk 1]
```diff
@@ -332,3 +332,18 @@ void CServiceBroker::UnregisterCPUInfo()
 {
   g_serviceBroker.m_cpuInfo.reset();
 }
+
+void CServiceBroker::RegisterTextureCache(const std::shared_ptr<CTextureCache>& cache)
+{
+  g_serviceBroker.m_textureCache = cache;
+}
```
Comment (fuzzard) [inline xbmc/ServiceBroker.cpp:339]: Should this possibly be 

```
void CServiceBroker::RegisterTextureCache(const std::shared_ptr<CTextureCache> cache)
{
  g_serviceBroker.m_textureCache = std::move(cache);
}
```

I thought i had seen some clang-tidy performance changes in the past and we seem to use that style below in https://github.com/xbmc/xbmc/blob/c041487397533b55bccdbad60a4b0ddc30766df4/xbmc/ServiceBroker.cpp#L326-L329

I'll let others comment on whats the preferred way unless i can find some docs that say one way or the other.

**Review by fuzzard — COMMENTED** [review]
Generally looks good. Register/unregister look to occur in the correct order/places.
A lot of files changed, so havent checked every file whether #include "servicebroker.h" was required, but saw you added in a few places, so i'll go with you've got it covered if jenkins doesnt fail.

My other comment was more a query. The way you use Keyboardlayoutmanager and applicationmessenger from the initial creation makes me think you would leave those 2 as is regardless.

File: xbmc/ServiceBroker.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/ServiceBroker.cpp:339]: There are many discussions in the net around byval+move vs. ref. There is no right and wrong, more or less personal preference. As far as I know not tackled by our coding guidelines. SO, some people use byval+move, others ref. I prefer the latter.

**Comment by ksooo** [comment]
If nobody objects I will merge this PR the next days.

**Review by DaveTBlake — APPROVED** [review]
Looks fine for the music code.

**Pull request merged**